src/DynamoDbQueryBuilder.php

```diff
@@ -446,4 +478,138 @@ public function setClient($client)
+     * @return $this
+     */
+    public function withoutGlobalScopes(array $scopes = null)
+    {
+        if (is_array($scopes)) {
+            foreach ($scopes as $scope) {
+                $this->withoutGlobalScope($scope);
+            }
+        } else {
+            $this->scopes = [];
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get an array of global scopes that were removed from the query.
+     *
+     * @return array
+     */
+    public function removedScopes()
+    {
+        return $this->removedScopes;
+    }
+
+    /**
+     * Apply the scopes to the Eloquent builder instance and return it.
+     *
+     * @return \Illuminate\Database\Eloquent\Builder|static
+     */
+    public function applyScopes()
+    {
+        if (! $this->scopes) {
+            return $this;
+        }
+
+        $builder = $this;
+
+        foreach ($this->scopes as $identifier => $scope) {
+            if (! isset($builder->scopes[$identifier])) {
+                continue;
+            }
+
+            $builder->callScope(function (DynamoDbQueryBuilder $builder) use ($scope) {
+                // If the scope is a Closure we will just go ahead and call the scope with the
+                // builder instance. The "callScope" method will properly group the clauses
+                // that are added to this query so "where" clauses maintain proper logic.
+                if ($scope instanceof Closure) {
+                    $scope($builder);
+                }
+
+                // If the scope is a scope object, we will call the apply method on this scope
+                // passing in the builder and the model instance. After we run all of these
+                // scopes we will return back the builder instance to the outside caller.
+                if ($scope instanceof Scope) {
+                    $scope->apply($builder, $this->getModel());
+                }
+            });
+        }
+
+        return $builder;
+    }
+
+    /**
+     * Apply the given scope on the current builder instance.
+     *
+     * @param  callable  $scope
+     * @param  array  $parameters
+     * @return mixed
+     */
+    protected function callScope(callable $scope, $parameters = [])
+    {
+        array_unshift($parameters, $this);
+
+        // $query = $this->getQuery();
+
+        // // We will keep track of how many wheres are on the query before running the
+        // // scope so that we can properly group the added scope constraints in the
+        // // query as their own isolated nested where statement and avoid issues.
+        // $originalWhereCount = is_null($query->wheres)
+        //             ? 0 : count($query->wheres);
+
+        // var_dump($scope);
+
+        $result = $scope(...array_values($parameters)) ?: $this;
+
+        // if (count((array) $query->wheres) > $originalWhereCount) {
+        //     $this->addNewWheresWithinGroup($query, $originalWhereCount);
+        // }
+
+        return $result;
+    }
```

Comment:
Most of that, except the var_dump was carried over from Eloquent... left it in place so it could be implemented eventually, and with the update this weekend to FilterExpressions implementing it may be necessary - I'll take a look.